CLI dev mode (#436)
scaffold-eth/scaffold-eth-2: contributors/DEVELOPER-GUIDE.md

The CLI tool source code can be found under `src/`. As you make changes in those files, the development server will compile the source and generate the script inside `bin/`.

To run the script you can simply run
```bash
yarn cli
```

This command will do two things:
1. Remove the folder `test-cli/` if it exists.
2. Run the CLI script using the folder `test-cli/` as the destination.

If you wanted to use a different name or destination, you could also run the CLI via node.
```bash
node path-to-this-repo/bin/create-dapp-se2.js [project-name --flags]
```

You can send any option or flag to the CLI command, except for the project name, which is hardcoded to be "test-cli" in the package.json script. For example, a handy command is `yarn cli -s` to tell the CLI to skip installing dependencies.

## Changes to the Resulting Project

The source files for the instance projects can be found under `templates/`. You'll see there are two folders there: `base/` and `extensions/`. The `base/` folder has the source files that will be present in all instances, whereas `extensions/` hold the source files that will be added or not to the instances based on the user choices within the CLI tool.

> It's highly recommended that you go through [RFC-extensions.md](contributors/RFC-extensions.md) to understand the template API to create extensions. We use a custom template API to allow extensions to modify any file inside the `templates/` folder. While flexible and powerful, it requires developers to understand how it works. It's JS based, so there's no new technology needed to understand and use it.

Author: ```suggestion
It's highly recommended that you go through [RFC-extensions.md](RFC-extensions.md) to understand the template API to create extensions. We use a custom template API to allow extensions to modify any file inside the `templates/` folder. While flexible and powerful, it requires developers to understand how it works. It's JS based, so there's no new technology needed to understand and use it.
```

> It's highly recommended that you go through RFC-extensions.md (TODO add link once published to the right folder) to understand the template API to create extensions. We use a custom template API to allow extensions to modify any file inside the `templates/` folder. While flexible and powerful, it requires developers to understand how it works. It's JS based, so there's no new technology needed to understand and use it.

Author: ```suggestion
It's highly recommended that you go through [RFC-extensions.md](https://github.com/scaffold-eth/scaffold-eth-2/blob/cli-dev-mode/contributors/RFC-extensions.md) to understand the template API to create extensions. We use a custom template API to allow extensions to modify any file inside the `templates/` folder. While flexible and powerful, it requires developers to understand how it works. It's JS based, so there's no new technology needed to understand and use it.
```

> It's highly recommended that you go through [RFC-extensions.md](https://github.com/scaffold-eth/scaffold-eth-2/blob/cli-dev-mode/contributors/RFC-extensions.md) to understand the template API to create extensions. We use a custom template API to allow extensions to modify any file inside the `templates/` folder. While flexible and powerful, it requires developers to understand how it works. It's JS based, so there's no new technology needed to understand and use it.

Author: ```suggestion
It's highly recommended that you go through [RFC-extensions.md](contributors/RFC-extensions.md) to understand the template API to create extensions. We use a custom template API to allow extensions to modify any file inside the `templates/` folder. While flexible and powerful, it requires developers to understand how it works. It's JS based, so there's no new technology needed to understand and use it.
```
